DigitalLearningSolutions.Web/Scripts/trackingSystem/reports.ts

```diff
@@ -68,6 +90,30 @@ function generateChart(request: XMLHttpRequest) {
         });
       }
     });
+
+  const dataPointErrorContainer = document.getElementById('activity-graph-data-error');
+  if (dataPointErrorContainer !== null) {
+    dataPointErrorContainer.hidden = true;
+    dataPointErrorContainer.style.display = 'none';
+  }
+  const chartContainer = document.getElementById('activity-graph');
+  if (chartContainer !== null) {
+    chartContainer.hidden = false;
+    chartContainer.style.display = 'block';
+  }
+}
+
+function displayTooManyDataPointsMessage() {
+  const dataPointErrorContainer = document.getElementById('activity-graph-data-error');

```

Comment:
Can we extract the two `getElementById` calls in this function (and the one above) into two functions of their own? It would also be worth extracting the id strings into constants at the top of the file and commenting that they need to match the elements in the view (as is done with the `_ActivityTable.cshtml` element ids).